buildings/gui/new_capture_source_area.py

```diff
@@ -0,0 +1,341 @@
+        self.setup_toolbar()
+
+        self.le_area_title.setDisabled(True)
+        self.le_external_id.setDisabled(True)
+        self.mcb_selection_layer.setDisabled(True)
+        self.mcb_selection_layer.setExceptedLayerList([self.capture_source_area])
+        self.rb_select_from_layer.setChecked(False)
+
+        self.capture_source_area.featureAdded.connect(self.creator_feature_added)
+        self.capture_source_area.featureDeleted.connect(self.creator_feature_deleted)
+        self.capture_source_area.geometryChanged.connect(self.creator_geometry_changed)
+
+        self.rb_select_from_layer.toggled.connect(self.rb_select_from_layer_clicked)
+
+        self.btn_save.clicked.connect(partial(self.save_clicked, commit_status=True))
+        self.btn_reset.clicked.connect(self.reset_clicked)
+        self.btn_exit.clicked.connect(self.exit_clicked)
+
+        QgsMapLayerRegistry.instance().layerWillBeRemoved.connect(self.layers_removed)
+
+        iface.actionToggleEditing().trigger()
+        iface.actionAddFeature().trigger()
+
+    @pyqtSlot(bool)
+    def rb_select_from_layer_clicked(self, checked):
+        if checked:
+            self.mcb_selection_layer.setEnabled(True)
+            iface.actionSelect().trigger()
+            if self.mcb_selection_layer.count() > 0:
+                self.current_layer = self.mcb_selection_layer.currentLayer()
+                iface.setActiveLayer(self.current_layer)
+                self.current_layer.selectionChanged.connect(self.current_layer_selection_changed)
+            self.mcb_selection_layer.layerChanged.connect(self.mcb_selection_layer_changed)
+        else:
+            self.mcb_selection_layer.layerChanged.disconnect(self.mcb_selection_layer_changed)
+            if self.mcb_selection_layer.count() > 0:
+                self.current_layer.removeSelection()
+                self.current_layer.selectionChanged.disconnect(self.current_layer_selection_changed)
+
+            self.mcb_selection_layer.setDisabled(True)
+            iface.actionAddFeature().trigger()
+            self.current_layer = None
+            iface.setActiveLayer(self.current_layer)
+
+    @pyqtSlot(QgsMapLayer)
+    def mcb_selection_layer_changed(self, current_layer):
+        if current_layer is None:
+            return
+        if self.current_layer is not None:
+            self.current_layer.removeSelection()
+            try:
+                self.current_layer.selectionChanged.disconnect(self.current_layer_selection_changed)
+            except TypeError:
+                pass
+        self.current_layer = current_layer
+        iface.setActiveLayer(current_layer)
+        self.current_layer.selectionChanged.connect(self.current_layer_selection_changed)
+
+    @pyqtSlot()
+    def current_layer_selection_changed(self):
+        selection = self.current_layer.selectedFeatures()
+        if (len(selection) > 1) or (len(selection) == 0):
+            self.le_external_id.setDisabled(True)
```

Comment:
Might be good to have an info messagebar here to let people know that they cannot select more than one feature.